Fix ESMFold API requests - disable SSL to avoid certificate errors

naailkhan28 opened the pull request:
Fixes #80  - now ESMFold API requests work and don't fail SSL certificate checks.

`ProteinCartography/esmfold_apiquery.py` has been updated to disable SSL when making the API request, and suppress the warnings from the `requests` module.

Tested using `python ProteinCartography/esmfold_apiquery.py -i 2PK8_monomer.fasta -o ./2PK8_esmfold.pdb` - outputs a PDB file as expected.

mezarque commented on ProteinCartography/esmfold_apiquery.py:
```diff
@@ -53,7 +54,7 @@ def post_esmfold_apiquery(fasta: str):
     result = session_with_retry().post(
         "https://api.esmatlas.com/foldSequence/v1/pdb/",
-        data=fasta,
+        data=fasta, verify=False
```
nit: formatting

```suggestion
        data=fasta, 
        verify=False,
```

mezarque commented on ProteinCartography/api_utils.py:
```diff
@@ -7,6 +7,7 @@
 from requests.packages.urllib3.util.retry import Retry
 from tests import artifact_generation_utils, mocks
 
+
```
Removing this should resolve linting issues.

```suggestion
```

mezarque commented on ProteinCartography/esmfold_apiquery.py:
```diff
@@ -123,7 +124,12 @@ def main():
+    #Ignore warnings when we make the ESMFold API request
+    #We're disabling SSL which `requests` will otherwise warn us about
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
```
From my understanding, this would ignore all warnings that occur within this block. 

If we're trying just to ignore SSL warnings, should we be more specific and use `urllib3.exceptions.InsecureRequestWarning` and/or `ssl.SSLErrorWarning`?

Ignoring these kinds of warnings also seems like it might be a security vulnerability.

@mertcelebi what do you think?

mertcelebi replied:
Yeah, this should explicitly ignore the `InsecureRequestWarning` and that should work. We still would like to surface other warnings to the user. 

If you import, `from requests.packages.urllib3.exceptions import InsecureRequestWarning` and change this line to `warnings.simplefilter("ignore", category=InsecureRequestWarning)`, that would work.

mezarque reviewed (changes requested):
Thanks for submitting this PR! I made some syntax suggestions/ suggested fixes. I'm going to ask @mertcelebi to chime in about how we should think about handling SSL warnings. I think it might not be ideal to ignore all warnings that emerge during the `esmfold_apiquery()` request.

keithchev commented on ProteinCartography/esmfold_apiquery.py:
```diff
@@ -53,7 +54,7 @@ def post_esmfold_apiquery(fasta: str):
     """
     result = session_with_retry().post(
```
Since `verify=False` is a bit of a security concern, I think it'd be good to add a comment here to explain that  this is (hopefully) a temporary fix and will be removed if/when the ESM folks renew their certs.

keithchev reviewed (changes requested):
Thanks for fixing this! About the formatting issues that @mezarque pointed out: we use `ruff` and `snakefmt` for formatting, which you can run using `make format` in your local repo. This is usually easier than fixing the formatting manually 😄

mertcelebi commented on ProteinCartography/esmfold_apiquery.py:
```diff
@@ -123,7 +124,12 @@ def main():
     output_file = args.output
-    esmfold_apiquery(input_file, output_file)
+
+    #Ignore warnings when we make the ESMFold API request
```
nit: if you could leave a space between # and the comments, that'd be great, thank you!

naailkhan28 commented:
I've added more commits - I've changed the `warnings.simplefilter()` to only include the SSL warning, and I've reformatted my files using the `make format` workflow as suggested. I've also added a comment indicating that this is (hopefully) a temporary fix.

Let me know if there's anything else :)

mertcelebi reviewed (approved):
Looks great. Thank you for the contribution!

(PS: On Twitter, we saw that you were having issues regarding BLAST with the pipeline, if you share more details in a GitHub issue so we can replicate, we'd be happy to look into and try to issue a fix).

keithchev commented:
Looks great! There's just one more tiny formatting issue - the comment on line 57 in `esmfold_apiquery.py` is too long (the max line length in 100 chars and unfortunately ruff doesn't automatically fix this). If you break that line into two lines, then that should get the CI checks passing. Thanks!

naailkhan28 commented:
> (PS: On Twitter, we saw that you were having issues regarding BLAST with the pipeline, if you share more details in a GitHub issue so we can replicate, we'd be happy to look into and try to issue a fix).

I'm currently having some issues replicating my own BLAST errors! Will investigate further.

I also had some ideas on avoiding BLAST searches but I'll post these in #67

mezarque reviewed (approved):
lgtm! Thanks for contributing this!
